play-json/shared/src/main/scala/JsResult.scala

```diff
@@ -197,8 +197,33 @@ sealed trait JsResult[+A] { self =>
 }
 
 object JsResult {
+  import scala.util.{ Failure, Try, Success }
   import play.api.libs.functional._
 
+  case class Exception(cause: JsError)
```

Comment:
Maybe this should be in `JsError`? i.e. `JsError.Exception` is an exception produced from a `JsError`.